flink/src/test/java/org/apache/iceberg/flink/data/TestRowDataProjection.java

```diff
@@ -0,0 +1,161 @@
+    );
+    Schema id = new Schema(
+        Types.NestedField.required(0, "id", Types.LongType.get())
+    );
+    Schema data = new Schema(
+        Types.NestedField.optional(1, "data", Types.StringType.get())
+    );
+    generateAndValidate(schema, id);
+    generateAndValidate(schema, data);
+  }
+
+  @Test
+  public void testEmptyProjection() {
+    Schema schema = new Schema(
+        Types.NestedField.required(0, "id", Types.LongType.get()),
+        Types.NestedField.optional(1, "data", Types.StringType.get())
+    );
+    generateAndValidate(schema, schema.select());
+  }
+
+  @Test
+  public void testRename() {
+    Schema schema = new Schema(
+        Types.NestedField.required(0, "id", Types.LongType.get()),
+        Types.NestedField.optional(1, "data", Types.StringType.get())
+    );
+
+    Schema renamed = new Schema(
+        Types.NestedField.required(0, "id", Types.LongType.get()),
+        Types.NestedField.optional(1, "renamed", Types.StringType.get())
+    );
+    generateAndValidate(schema, renamed);
+  }
+
+  @Test
+  public void testNestedProjection() {
+    Schema schema = new Schema(
+        Types.NestedField.required(0, "id", Types.LongType.get()),
+        Types.NestedField.optional(3, "location", Types.StructType.of(
+            Types.NestedField.required(1, "lat", Types.FloatType.get()),
+            Types.NestedField.required(2, "long", Types.FloatType.get())
+        ))
+    );
+
+    // Project id only.
+    Schema idOnly = new Schema(
+        Types.NestedField.required(0, "id", Types.LongType.get())
+    );
+    generateAndValidate(schema, idOnly);
+
+    // Project lat only.
+    Schema latOnly = new Schema(
+        Types.NestedField.optional(3, "location", Types.StructType.of(
+            Types.NestedField.required(1, "lat", Types.FloatType.get())
+        ))
+    );
+    generateAndValidate(schema, latOnly);
+
+    // Project long only.
+    Schema longOnly = new Schema(
+        Types.NestedField.optional(3, "location", Types.StructType.of(
+            Types.NestedField.required(2, "long", Types.FloatType.get())
+        ))
+    );
+    generateAndValidate(schema, longOnly);
+
+    // Project location.
+    Schema locationOnly = schema.select("location");
+    generateAndValidate(schema, locationOnly);
+  }
+
+  private void generateAndValidate(Schema schema, Schema projectSchema) {
+    int numRecords = 100;
+    Iterable<Record> recordList = RandomGenericData.generate(schema, numRecords, 102L);
+    Iterable<RowData> rowDataList = RandomRowData.generate(schema, numRecords, 102L);
+
+    StructProjection structProjection = StructProjection.create(schema, projectSchema);
+    RowDataProjection rowDataProjection = RowDataProjection.create(schema, projectSchema);
+
+    Iterator<Record> recordIter = recordList.iterator();
+    Iterator<RowData> rowDataIter = rowDataList.iterator();
+
+    for (int i = 0; i < numRecords; i++) {
+      Assert.assertTrue("Should have more records", recordIter.hasNext());
```

Comment:
I am not sure it is necessary to have the assertions on line 149, 150, 158, 159. this is like to test the `RandomGenericData.generate` function.

even if we want to do, we can keep recordList and rowDataList are as `List`. Then we can just verify the size of the  list